### DISCUSSION TITLE
Retry failed publish to AUR

### DISCUSSION THREAD

**Discussion by Zebradil** [goreleaser/goreleaser#4724] [Q&A] [answered]
I publish with goreleaser to multiple registries, including AUR. There was an issue with AUR some day and the release failed partially: it was uploaded to the container image registry and to the Github release but not to the AUR.

```
  ⨯ release failed after 4m25s               error=1 error occurred:
	* arch user repositories: git: failed to clone local repository: "clone ssh://aur@aur.archlinux.org/cloudflare-dynamic-dns-bin.git cloudflare-dynamic-dns-bin-master" failed: Cloning into 'cloudflare-dynamic-dns-bin-master'...
kex_exchange_identification: read: Connection reset by peer
Connection reset by 95.216.144.15 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

And now I'm not sure how to retry publishing only to AUR. It still tries to upload artifacts to the Github release (and fails, now because the artifacts were already uploaded there), if I run `goreleaser release --skip announce,before,docker,homebrew,ko,nix,sbom,scoop,sign,snapcraft,validate,winget --clean` (these are all available skips, except aur and publish).

It seems like there are two points to address here:

1. Provide a way to publish only to selected registry (e.g. only to AUR).
2. Retry publishing to AUR as it is done with docker registries. There was a similar question about retries when publishing container images: https://github.com/orgs/goreleaser/discussions/4263

If the second point makes sense, I can implement a PR similar to #4265.

**Comment by caarlos0** [comment, marked as answer]
I think currently the easiest way is to grab the generated files from the dist folder and push them manually 

we should retry it, though, yes.

**Reply by Zebradil** [reply]
> push them manually

Yes, it's an option. Or maybe to create a no-op patch release.

But what about retries: does it make sense, should I submit a PR?

**Reply by caarlos0** [reply]
I think it does, yes... feel free to!

thanks 🙏

**Reply by Zebradil** [reply]
I checked the code and decided to add retries not to the aur-specific code but to the git clone part. Please, take a look: #4725

**Reply by caarlos0** [reply]
thanks, merged!
